rust/chains/hyperlane-ethereum/src/rpc_clients/fallback.rs

```diff
@@ -119,15 +265,18 @@ impl JsonRpcClient for FallbackProvider<PrometheusJsonRpcClient<Http>> {
             if !errors.is_empty() {
                 sleep(Duration::from_millis(100)).await;
             }
-            for (idx, provider) in self.0.iter().enumerate() {
+            let priorities_snapshot = take_priorities_snapshot(self).await;
+            for (idx, priority) in priorities_snapshot.iter().enumerate() {
+                let provider = &self.inner.providers[priority.index];
                 let fut = match params {
                     Value::Null => provider.request(method, ()),
                     _ => provider.request(method, &params),
                 };
-
                 let resp = fut.await;
+                handle_stalled_provider(self, priority).await?;
                 let _span =
                     warn_span!("request_with_fallback", provider_index=%idx, ?provider).entered();
```

Comment:
before the `provider_index` meant two things - the first being to indicate the # of providers that had been tried previously when falling back, and the second being which provider in the list of providers was currently being used

now that things have changed a bit, wondering if we should change this to something like `fallback_count=%idx`? and if we keep `provider_index`, to have it be `provider_index=%priority.index` or something of the like